rakefile

```diff
@@ -9,6 +11,56 @@ include PerfResultsHelper
 
 REF_ARCH_STD, REF_ARCH_LARGE = ["S","L"]
 
+# Create a trumped up class to get a Beaker helpers outside of a Beaker run
+class ClassMixedWithDSLHelpers
+  include Beaker::DSL::Helpers
+  include Beaker::DSL::Wrappers
+  include Beaker::DSL::Roles
+  include Beaker::DSL::Patterns
+
+  def options
+    { cache_files_locally: true }
+  end
+
+  def logger
+    Beaker::Logger.new
+  end
+end
+
+def master_from_beaker_hosts_hash(hosts)
+  master = nil
+  hosts.each_key do |name|
+    host = hosts[name]
+    master = host if host[:roles].include?("master")
+  end
+  master
+end
+
+def fetch_pe_for_master(beaker_proxy, master, ver, dir, dest = "/tmp")
+  local = File.directory?(dir)
+  filename = "puppet-enterprise-#{ver}-#{master['packaging_platform']}"
+  if local
+    extension = File.exist?("#{dir}/#{filename}.tar.gz") ? ".tar.gz" : ".tar"
+    non_exist_msg = "PE #{dir}/#{filename}#{extension} does not exist"
+    raise non_exist_msg unless File.exist?("#{dir}/#{filename}#{extension}")
+
+  else
+    extension = beaker_proxy.link_exists?("#{dir}/#{filename}.tar.gz") ? ".tar.gz" : ".tar"
```

Comment:
no idea what this beaker_proxy is or what it is doing.  Some commentary could help.

Reply:
A beaker class instance is needed to gain access to the body of beaker helper methods.  The custom class at the top of the Rakefile is the best solution I could come up with for this.

Reply:
yeah, but what is beaker proxy?  My guess is that this line is somehow checking if the link exists on the master?  But I am not sure.

Reply:
I have updated the references.  Proxy was a misnomer.  This line specifically ensures that the given pe package exists at the pe_dir provided at the remote location.  I have added a comment to this effect.